spyder/plugins/explorer/widgets/explorer.py

```diff
@@ -558,6 +648,80 @@ def open(self, fnames=None):
             else:
                 self.open_outside_spyder([fname])
 
+    @Slot()
+    def open_association(self, app_path):
+        """Open files with given application executable path."""
+        return_codes = {}
+        if app_path:
+            fnames = self.get_selected_filenames()
+
+            # Add quotes as needed
+            if sys.platform != 'darwin':
+                new_fnames = []
+                for fname in fnames:
+                    if ' ' in fname:
+                        if '"' in fname:
+                            fname = '"{}"'.format(fname)
+                        else:
+                            fname = "'{}'".format(fname)
+                        new_fnames.append(fname)
+                    else:
+                        new_fnames.append(fname)
+                fnames = new_fnames
+
+            if sys.platform == 'darwin':
+                cmd = ['open', '-a', app_path] + fnames
+                return_code = subprocess.call(cmd)
+                return_codes[' '.join(cmd)] = return_code
+            elif os.name == 'nt':
+                cmd = [app_path] + fnames
+                return_code = subprocess.call(cmd)

```

Comment:
Does this open a CMD window for a little bit or not? I say it because that's usually the case with the `subprocess` calls on Windows.

Reply:
On the windows tests (and the demo I did last week) I did not see the black dialog but I will double check now